### PR TITLE
Fix/wrong file extention error handling

### PR DESCRIPTION
**Pull request by dbshah1212**
# Description of change
- Handled empty CSV and wrong formated files.

# Manual QA steps
 - Created empty csv, Jsonl with csv format, zip with jsonl format, empty jsonl file and csv with jsonl format, and all errors handled so gracefully. 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_s3_csv/s3.py [hunk 1]
```diff
@@ -230,6 +239,7 @@ def sampling_gz_file(table_spec, s3_path, file_handle, sample_rate):
 def sample_file(table_spec, s3_path, file_handle, sample_rate, extension):
     global skipped_files_count
 
+    global skipped_files_count
```
Comment (karanpanchal-crest) [inline tap_s3_csv/s3.py:242]: @dbshah1212 Why are we declaring same variable twice?

File: tap_s3_csv/sync.py [hunk 2]
```diff
@@ -61,13 +61,15 @@ def sync_table_file(config, s3_path, table_spec, stream):
...
         s3.skipped_files_count = s3.skipped_files_count + 1
         return 0
-    if extension == "zip":
-        return sync_compressed_file(config, s3_path, table_spec, stream)
-    if extension in ["csv", "gz", "jsonl", "txt"]:
-        return handle_file(config, s3_path, table_spec, stream, extension)
-
-    LOGGER.warning('"%s" having the ".%s" extension will not be synced.',s3_path,extension)
-    s3.skipped_files_count = s3.skipped_files_count + 1
+    try:
```
Comment (karanpanchal-crest) [inline tap_s3_csv/sync.py:64]: @dbshah1212 Can you please provide a scenario in the comment about the UnicodeDecodeError and JSONDecodeError exceptions

File: tap_s3_csv/s3.py (hunk 1 above)
Comment (dbshah1212) [reply tap_s3_csv/s3.py]: It left while resolving merge conflict, removed now.

File: tap_s3_csv/sync.py (hunk 2 above)
Comment (dbshah1212) [reply tap_s3_csv/sync.py:64]: Added in except clause.

File: tests/unittests/test_comporessed_file.py [hunk 3]
```diff
@@ -432,6 +508,29 @@ def test_syncing_of_unsupported_file_records(self, mocked_logger):
...
+        mocked_logger.assert_called_with('Skipping %s file as parsing failed. Verify an extention of the file.',s3_path)
+        self.assertEqual(0, records)
+
+    @mock.patch("tap_s3_csv.sync.handle_file", side_effect=mock_jsonl_sample_file)
+    def test_syncing_of_gz_file_with_jsonl_extantion(self, mock_jsonl_sample_file,mocked_logger):
+        config = {"bucket" : "bucket_name"}
+        table_spec = { "table_name" : "GZ_DATA"}
+        stream = {}
+        s3_path = "unittest_compressed_files/gz_stored_as_jsonl.jsonl"
+        extension = "jsonl"
```
Comment (karanpanchal-crest) [inline tests/unittests/test_comporessed_file.py:529]: @dbshah1212 What is the use of the 'extension' variable??

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KAllan357 — APPROVED** [review]

**Pull request merged**